Feature/47 Password recovery refactoring

sanioooook opened the pull request.

sanioooook committed:
moved the functionality to the module

sanioooook committed:
Password recovery refactoring

horba commented on front/src/components/recovery-password/mmt-recovery-step-1.vue:
```diff
@@ -0,0 +1,11 @@
+<template src='./mmt-recovery-step-1.html'>
+
+</template>
```
one line

sanioooook replied:
okey

horba commented on front/src/components/recovery-password/mmt-recovery-step-1.vue:
```diff
@@ -0,0 +1,11 @@
+
+</script>
+
+<style scoped src="./mmt-recovery-step.css">
```
why css without step numbers?

sanioooook replied:
because there are only two styles and they only touch two steps. the rest is governed by the built-in classes

horba commented on front/src/components/recovery-password/mmt-recovery-step-2.js:
```diff
@@ -0,0 +1,39 @@
+          password: this.passwordConfirmation,
+          token: this.$route.params.token
+        }).then(this.step = 3).catch(() => {
+        this.step = 2;
```
i can\t understand why do we need to know number of step here

horba commented on front/src/store/modules/recoveryPassword.js:
```diff
@@ -0,0 +1,14 @@
+      return axios.post(`${serverPath}/api/forgotPassword/resetPassword/`, payload);
+    }
+  }
+};
```
something wrong with style of file naming

horba commented on front/src/views/recovery-password/recovery-password.view.js:
```diff
@@ -1,47 +1,21 @@
-        samePass: () => this.password === this.passwordConfirmation
-          || this.$t('recoveryPassword.passwordMustIdentity')
-      }
+      step: 1
```
hm...i don't think that it is vue way. try to add custom event for your step components and handlers for it.

like 
`<mmt-recovery-step-1 @stepDone="handleStep({stepNumber: 1})"`
`<mmt-recovery-step-2 @stepDone="handleStep({stepNumber: 2})"`

horba replied:
just think,  might some events or properties could make your code easier

sanioooook committed:
correction of comments

sanioooook committed:
correction of comments

sanioooook committed:
Merge branch 'dev' of https://github.com/horba/my-trip into feature/47

horba commented on server/WebAPI/Controllers/RecoveryPasswordController.cs:
```diff
@@ -0,0 +1,46 @@
+      {
+        return Ok(await recoveryPasswordService.SendEmailAsync(recoveryPasswordModel));
+      }
+      else
```
https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-2.1

horba commented on server/WebAPI/Controllers/RecoveryPasswordController.cs:
```diff
@@ -0,0 +1,46 @@
+      if(recoveryPasswordModel.Email != "" &&
+        recoveryPasswordModel.Password != "" &&
+        recoveryPasswordModel.Token != "")
+      {
```
https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-2.1

horba commented on front/src/views/recovery-password/recovery-password.view.html:
```diff
@@ -31,130 +31,13 @@ <h4>{{$t('recoveryPassword.title')}}</h4>
-            </v-row>
-          </v-form>
-        </v-card>
+          <mmt-recovery-step-2 @stepDone="step = 3" @stepNotDone="step = 2"/>
```
- currentStep
- js handle function could be better than template code
- stepDone and stepNotDone confuses me. it is more appreciate  to use just stepDone for each step component
- might, it is time to understand presentation and container components. For example, your step components are only presentation, and your recovery-password is container. it is just thought. read article and decide. 

https://medium.com/@kanby/%D0%BF%D1%80%D0%B5%D0%B7%D0%B5%D0%BD%D1%82%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9-%D0%BA%D0%BE%D0%BC%D0%BF%D0%BE%D0%BD%D0%B5%D0%BD%D1%82-%D0%B8-%D0%BA%D0%BE%D0%BD%D1%82%D0%B5%D0%B9%D0%BD%D0%B5%D1%80-%D0%B2-react-f0e118480809

sanioooook committed:
correcting path names
deleting stepNotDone
deleting step
adding a new password update model
deleting an extra model
adding asynchronous calls instead of synchronous ones

sanioooook committed:
corrected the style of the input field

sanioooook committed:
Merge branch 'dev' of
https://github.com/horba/my-trip into feature/47

sanioooook committed:
rename recoveryPassword file module
